web/src/App.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+  Box,
+  Center,
+  Flex,
+  Text,
+  useColorMode,
+  useToast,
+} from "@chakra-ui/react";
+import * as React from "react";
+import {
+  Navigate,
+  Route,
+  Routes,
+  useLocation,
+  useNavigate,
+} from "react-router-dom";
+import { useRecoilState, useRecoilValue } from "recoil";
+
+import { useAnalytics } from "@/analytics";
+import { Loader } from "@/components/customcomponents/loader/Loader";
+import { Footer } from "@/components/Footer";
+import { Nav } from "@/components/navBar/Nav";
+import { AnalyticsDashboard } from "@/pages/Analytics";
+import { Overview } from "@/pages/Configure";
+import { NotificationsMap } from "@/pages/Dashboard";
+import { HomePage } from "@/pages/HomePage";
+import { useConnectionState } from "@/view/hooks/hooks";
+
+import { redirectToHomaPage } from "./data/recoil";
+
+const PrivateRoute: React.FC<{ children: React.ReactElement }> = ({
+  children,
+}) => {
+  // Private routes will ensure user is connected to singlestore before using the route.
+  // Will redirect to connection page in case user in not connected.
+  const redirect = useRecoilValue(redirectToHomaPage);
+  if (redirect) {
+    return <Navigate to="/" />;
+  }
+  return children;
+};
+
+const LayoutContainer = ({ children }: { children: React.ReactElement }) => {
+  const redirect = useRecoilValue(redirectToHomaPage);
```

Comment:
There will happen two issues in your approach you can try

1. If you make even a single change in configure page credentials it will immediately redirect to home page which it should not

2. I need to now have nav and footer in each router which i think is not a optimal way to integrate nav and footer. It should be rendered once and remain consistent throughout even if anything happens to inner components.

also, as discussed earlier. If i don't have a redirect I cannot redirect to the actual page user was trying to open which I think is fine and can avoid but It already solved so should we keep it ? It also makes link shareable else user cannot share a specific page in the website

Reply:
Can Ignore

I retried with very minor tweeks from the commit 

worked like a charm